jgrapht-core/src/test/java/org/jgrapht/graph/AsSynchronizedGraphTest.java

```diff
@@ -0,0 +1,502 @@
+        order2.clear();
+        createOrder(order2, 10, 20, false); // rm 39 edges
+        order3.clear();
+        for (int i = 3; i < 15; i++)    // rm 12 vertices, 0 edges
+            order3.add(new RmV(i));
+        for (int i = 30; i < 40; i++)   // rm 10 vertices
+            order3.add(new RmV(i));
+        ts = new ActiveTestSuite();
+        ts.addTest(new TestThread("thread1"));
+        ts.addTest(new TestThread("thread2"));
+        ts.addTest(new TestThread("thread3"));
+        TestRunner.run(ts);
+        assertEquals(38, g.vertexSet().size());
+        assertEquals(190, g.edgeSet().size());
+        assertEquals(0, g.edgesOf(2).size());
+        assertEquals(19, g.edgesOf(41).size());
+
+    }
+    private void createOrder(ArrayList<Order> list, int start, int end, boolean add)
+    {
+        for (int i = start; i < end - 1; i++) {
+            for (int j = i + 1; j < end; j++) {
+                if (add)
+                    list.add(new AddE(i, j, new DefaultEdge()));
+                else
+                    list.add(new RmE(i, j));
+            }
+        }
+    }
+    private interface Order
+    {
+        void execute();
+    }
+    private class AddV
+        implements Order
+    {
+        int vertex;
+        public AddV(int v)
+        {
+            vertex = v;
+        }
+
+        @Override
+        public void execute() {
+            g.addVertex(vertex);
+        }
+    }
+    private class AddE
+        implements Order
+    {
+        DefaultEdge e;
+        int s, t;
+        public AddE(int s, int t, DefaultEdge e)
+        {
+            this.e = e;
+            this.s = s;
+            this.t = t;
+            this.e.source = s;
+            this.e.target = t;
+        }
+
+        @Override
+        public void execute()
+        {
+            g.addEdge(s, t, e);
+        }
+    }
+    private class SetCache
+        implements Order
+    {
+        @Override
+        public void execute()
+        {
+            g.setCache(!g.isCacheEnabled());
+        }
+    }
+    private class RmV
+        implements Order
+    {
+        int v;
+        public RmV(int v)
+        {
+            this.v = v;
+        }
+
+        @Override
+        public void execute()
+        {
+            g.removeVertex(v);
+        }
+    }
+    private class RmE
+        implements Order
+    {
+        int s, t;
+        public RmE(int s, int t)
+        {
+            this.s = s;
+            this.t = t;
+        }
+
+        @Override
+        public void execute()
+        {
+            g.removeEdge(s, t);
+        }
+    }
+    private int iteratorCnt(Iterator it)
+    {
+        int count = 0;
+        while (it.hasNext()) {
+            it.next();
+            count++;
+        }
+        return count;
+    }
+    public class TestThread
+            extends TestCase
+    {
+        public TestThread(String s)
+        {
+            super(s);
+        }
+        public void vertex()
```

Comment:
Where does this vertex() method ever get called?  Likewise for addEdge(), removeVertex(), removeEdge().  Is there some by-name reflective execution going on that I missed?  If not, please delete dead code.